### PR TITLE
Support projects' namespaces in IDE

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description
This PR is a support for projects' namespaces, added to Enso Engine in commits faf4214b2998776e16988bf1c98a6a5a658024f0 and 334a022ffd8191603c5d709d10fba40a2bf986a7

To test it, you need an engine 0.2.13-SNAPSHOT.2021-06-30 or later.

### Important Notes
Each project name is assisted with a namespace name: all changes are made to reflect that fact in IDE.

### Checklist
Please include the following checklist in your PR:

- [ ] The `CHANGELOG.md` was updated with the changes introduced in this PR.
- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [x] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [ ] ~~All code has been manually tested in the "debug/interface" scene.~~
- [x] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/rust/ide/src/double_representation/module.rs [hunk 1]
```diff
@@ -147,23 +148,23 @@ impl Id {
 #[serde(into="String")]
 #[serde(try_from="String")]
 pub struct QualifiedName {
-    /// The first segment in the full qualified name.
-    pub project_name : ReferentName,
+    /// The first segment in the full qualified name. May be a project name or a keyword like
```
Comment (mwu-tow) [inline src/rust/ide/src/double_representation/module.rs]: > May be a project name or a keyword like
 
 This is not clear and looks as unfinished.

File: src/rust/ide/src/double_representation/module.rs [hunk 2]
```diff
@@ -488,6 +493,8 @@ impl Info {
     /// Add a new import if the module is not already imported.
     pub fn add_module_import
     (&mut self, here:&QualifiedName, parser:&parser::Parser, to_add:&QualifiedName) {
+        let imports = self.iter_imports().collect_vec();
+        DEBUG!("add_module_import: {to_add} in {imports:?}");
```
Comment (mwu-tow) [inline src/rust/ide/src/double_representation/module.rs]: Should be removed or use logger.

File: src/rust/ide/src/double_representation/module.rs [hunk 3]
```diff
@@ -732,14 +739,14 @@ mod tests {
...
+        assert!(QualifiedName::try_from("namespace.project.Name").is_err());
+        assert!(QualifiedName::try_from("namespace.Project.name").is_err());
+        assert!(QualifiedName::try_from("namespace.").is_err());
         assert!(QualifiedName::try_from(".Name").is_err());
         assert!(QualifiedName::try_from(".").is_err());
         assert!(QualifiedName::try_from("").is_err());
-        assert!(QualifiedName::try_from("Project.Name").is_ok());
-        assert!(QualifiedName::try_from("Project.Name.Sub").is_ok());
+        assert!(QualifiedName::try_from("namespace.Project.Name").is_ok());
+        assert!(QualifiedName::try_from("namespace.Project.Name.Sub").is_ok());
```
Comment (mwu-tow) [inline src/rust/ide/src/double_representation/module.rs:749]: No cases for upper-cased namespace.

File: src/rust/ide/src/model/project.rs [hunk 4]
```diff
@@ -27,7 +30,10 @@ use uuid::Uuid;
 pub trait API:Debug {
     /// Project's name
     // TODO [mwu] This should return Rc<ReferentName>.
-    fn name(&self) -> ImString;
+    fn name(&self) -> ReferentName;
```
Comment (mwu-tow) [inline src/rust/ide/src/model/project.rs:32]: If it returns the ReferentName, the TODO should be removed.

**Review by mwu-tow — APPROVED** [review]
LGTM

**Comment by farmaazon** [comment]
The work was included in #1710 and merged with that PR,

**Pull request closed without merging**